tools/ignore_pot_creation_date.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import logging
+import os
+import subprocess
+import sys
+import typing
+
+"""
+This script reverts changes in PO files when only the timestamp has changed.
+It helps to create meaningful commits by ignoring unnecessary changes.
+"""
+
+
+def get_git_changeset(from_ref, to_ref) -> str:
+    """
+    Constructs the changeset string for `git diff` based on from_ref and
+    to_ref.
+    """
+    from_ref = (
+        from_ref
+        or os.getenv("PRE_COMMIT_FROM_REF")
+        or os.getenv("PRE_COMMIT_SOURCE")
+        or "HEAD"
+    )
+    to_ref = (
+        to_ref
+        or os.getenv("PRE_COMMIT_TO_REF")
+        or os.getenv("PRE_COMMIT_ORIGIN")
+    )
+    return f"{from_ref}...{to_ref}" if to_ref else from_ref
+
+
+def count_meaningful_changes(changeset, po_file) -> int:
+    """
+    Counts meaningful changes in the diff for a given PO file.
+    """
+    cmd = ["git", "diff", "--patch", "--unified=0", changeset, "--", po_file]
+    proc = subprocess.run(cmd, capture_output=True, text=True)
+    proc.check_returncode()
+
+    diff_lines = proc.stdout.splitlines()
+    return sum(
+        1
+        for line in diff_lines
+        if (line.startswith("-") or line.startswith("+"))
+        and "POT-Creation-Date:" not in line
+        and "PO-Revision-Date:" not in line
+        and po_file not in line
+    )
+
+
+def revert_po_file(changeset, po_file, logger):
+    """
+    Reverts a PO file to its original state in the given changeset.
+    """
+    ref = changeset.split("...")[
+        0
+    ]  # Use the first part of the changeset as the reference
+    logger.info(f"{po_file} has no meaningful changes, reverting to {ref}")
+    cmd = ["git", "show", f"{ref}:{po_file}"]
+    proc = subprocess.run(cmd, capture_output=True, text=True)
+    proc.check_returncode()
+
+    with open(po_file, "w") as file:
```

Comment:
if `po_file` is a Path, you can do this:
```suggestion
    with po_file.open(mode="w") as file:
```